plan/types.go

```diff
@@ -67,6 +67,47 @@ func (plan *PlanSpec) Replace(root, with PlanNode) {
 	plan.Roots[with] = struct{}{}
 }
 
+// CheckIntegrity checks the integrity of the plan, i.e., if node A is predecessor of B, then B must be successor of A.
+// This check only detects this problem (N2 is predecessor of R, but not successor):
+//
+//     N1 <----> R
+//               |
+//     N2 <-------
+//
+// But not this one (N2 is not predecessor of R, and, thus, it won't be visited):
+//
+//     N1 <-------
+//               |--> R
+//     N2 --------
+//
+// Nor this one (R2 is only successor of N1, and, thus, it won't be visited):
+//
+//     N1 <----> R1
+//     |
+//     --------> R2
+//
```

Comment:
Should we perform deeper checks and cover the cases specified in the function doc?  
We could also check for cycles.